mdp/linear_flows_online.py

```diff
@@ -0,0 +1,607 @@
+from builtins import str
+import mdp
+from mdp import numx
+from .linear_flows import FlowException, FlowExceptionCR, _sys, _traceback
+from collections import deque as _deque
+
+
+class OnlineFlowException(mdp.MDPException):
+    """Base class for exceptions in Flow subclasses."""
+    pass
+
+
+class OnlineFlow(mdp.Flow):
+    """An 'OnlineFlow' is a sequence of online/executable nodes that are trained and executed
+    together to form a more complex algorithm.  Input data is sent to the
+    first node and is successively processed by the subsequent nodes along
+    the sequence.
+
+    Using an online flow as opposed to handling manually a set of nodes has a
+    clear advantage: The general online flow implementation automatizes the
+    training (including supervised training and multiple training phases),
+    execution, and inverse execution (if defined) of the whole sequence.
+
+    OnlineFlow sequence can contain trained or non-trainable Nodes and optionally
+    a terminal trainable Node.
```

Comment:
Please let me know if this makes sense. Then I can improve the doc string of the Online flow to disambiguate some of the terms used.